app/models/subject.rb

```diff
@@ -71,4 +71,23 @@ def retired_for_workflow?(workflow)
       false
     end
   end
+
+  def surrounding(subject_set_id, window=5, gap=1)
+    all_smses = SetMemberSubject.where(subject_set_id: subject_set_id).order(:priority)
+    this_sms = all_smses.where(subject_id: id).first
+    this_index = all_smses.find_index(this_sms)
```

Comment:
I think we pass the current SMS record / priority in as we do know where are in the stack already.